QualifiedTeachersApi/src/QualifiedTeachersApi/V3/Handlers/GetQtsCertificateHandler.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using PdfSharpCore.Pdf;
+using PdfSharpCore.Pdf.AcroForms;
+using PdfSharpCore.Pdf.IO;
+using QualifiedTeachersApi.DataStore.Crm;
+using QualifiedTeachersApi.DataStore.Crm.Models;
+using QualifiedTeachersApi.V3.Requests;
+
+namespace QualifiedTeachersApi.V3.Handlers;
+
+public class GetQtsCertificateHandler : IRequestHandler<GetQtsCertificateRequest, FileResult>
+{
+    private const string QtsFormNameField = "Full Name";
+    private const string QtsFormTrnField = "TRN";
+    private const string QtsFormDateField = "QTSDate";
+
+    private readonly IDataverseAdapter _dataverseAdapter;
+    private readonly BlobServiceClient _blobServiceClient;
+
+    public GetQtsCertificateHandler(
+        IDataverseAdapter dataverseAdapter,
+        BlobServiceClient blobServiceClient)
+    {
+        _dataverseAdapter = dataverseAdapter;
+        _blobServiceClient = blobServiceClient;
+    }
+
+    private FileResult GetPdfStream(PdfDocument pdf)
+    {
+        byte[] pdfData;
+        using (var stream = new MemoryStream())
+        {
+            pdf.Save(stream, false);
+            pdfData = stream.ToArray();
+        }
+
+        return new FileContentResult(pdfData, "application/pdf")
+        {
+            FileDownloadName = "QTS Certificate.pdf"
+        };
+    }
+
+    public async Task<FileResult> Handle(GetQtsCertificateRequest request, CancellationToken cancellationToken)
```

Comment:
Could we return `Stream` here (or `byte[]`) instead of `FileResult`? Generally we try to keep these handlers decoupled from MVC types.